flash/image/classification/data.py

```diff
@@ -53,6 +58,65 @@ def load_sample(self, sample: Dict[str, Any], dataset: Optional[Any] = None) ->
+        predict_file: Optional[str] = None,
+        predict_images_root: Optional[str] = None,
+        predict_resolver: Optional[Callable[[str, str], str]] = None,
+        **data_source_kwargs: Any,
+    ) -> Any:
+        """Creates a :class:`~flash.image.classification.data.ImageClassificationData` object from the given CSV
+        files using the :class:`~flash.core.data.data_source.DataSource` of name
+        :attr:`~flash.core.data.data_source.DefaultDataSources.CSV` from the passed or constructed
+        :class:`~flash.core.data.process.Preprocess`.
+
+        Args:
+            input_field: The field (column) in the CSV file to use for the input.
+            target_fields: The field or fields (columns) in the CSV file to use for the target.
+            train_file: The CSV file containing the training data.
+            train_images_root: The directory containing the train images. If ``None``, the directory containing the
+                ``train_file`` will be used.
+            train_resolver: The function to use to resolve filenames given the ``train_images_root`` and IDs from the
+                ``input_field`` column.
+            val_file: The CSV file containing the validation data.
+            val_images_root: The directory containing the validation images. If ``None``, the directory containing the
+                ``val_file`` will be used.
+            val_resolver: The function to use to resolve filenames given the ``val_images_root`` and IDs from the
+                ``input_field`` column.
+            test_file: The CSV file containing the testing data.
+            test_images_root: The directory containing the test images. If ``None``, the directory containing the
+                ``test_file`` will be used.
+            test_resolver: The function to use to resolve filenames given the ``test_images_root`` and IDs from the
+                ``input_field`` column.
+            predict_file: The CSV file containing the data to use when predicting.
+            predict_images_root: The directory containing the predict images. If ``None``, the directory containing the
+                ``predict_file`` will be used.
+            predict_resolver: The function to use to resolve filenames given the ``predict_images_root`` and IDs from
+                the ``input_field`` column.
+            data_source_kwargs: Additional keyword arguments to use when constructing the data source.
+
+        Returns:
+            The constructed data module.
+        """
+        return cls(**data_source_kwargs).to_datasets(
+            (train_file, input_field, target_fields, train_images_root, train_resolver),
+            (val_file, input_field, target_fields, val_images_root, val_resolver),
+            (test_file, input_field, target_fields, test_images_root, test_resolver),
+            (predict_file, input_field, target_fields, predict_images_root, predict_resolver),
```

Comment:
I just get the sense there has to be a better way to do this, the number of arguments to these DataSource/Module functions is pretty huge and doesn't give a very good developer experience

Reply:
I am curious, what is your idea ?

Reply:
Well at least at first glance, there are a lot of repeated arguments (input field, target fields) and train/test/val/predict are repeated all over the place in this codebase. 

A module abstracting this series could be created so if there was a function like this where train/test/val are all in a series that module could be used to look up arguments. 

```
class FlashArgGroup:

flashArgGroup.predict.input_field to get a specific value from predict/test/train, would work like a namespace from argparse
flashArgGroup.test # to get all arguments as kwargs for the testing stage
```

Various function interfaces could be designed around that paradigm and all these functions that need a ton of arguments could just handle it with that interface.

That's just a first thought though, I'm sure there are simpler ways.